include/clap/ext/draft/midi-info.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include "../../plugin.h"
+
+// This extension lets a plugin expose it's MIDI implementation.
+// The primary goal is to allow a host to create a MIDI-CI Property Exchange "ChCtrlList" resource for the plugin,
+// so keyboard controllers can assign their knobs to suitable controllers.
+
+//TODO : clap_midiinfo could contain more (optional) info (min/max, stepcount, paramPath etc.)
+//TODO : for completeness this should work per-input port. Is that worth it?
+
+static CLAP_CONSTEXPR const char CLAP_EXT_MIDIINFO[] = "clap.midi-info.draft/0";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+   //MIDI 2.0 message status
+   CLAP_MIDIINFO_STATUS_REGISTERED_PER_NOTE_CONTROLLER = 0x00,
+   CLAP_MIDIINFO_STATUS_ASSIGNABLE_PER_NOTE_CONTROLLER = 0x10,
+   CLAP_MIDIINFO_STATUS_RPN = 0x20,
+   CLAP_MIDIINFO_STATUS_NRPN = 0x30,
+   CLAP_MIDIINFO_STATUS_PER_NOTE_PITCHBEND = 0x60,
+   CLAP_MIDIINFO_STATUS_POLY_AFTERTOUCH = 0xA0,
+   CLAP_MIDIINFO_STATUS_CONTROLLER = 0xB0,
+   CLAP_MIDIINFO_STATUS_CHANNEL_AFTERTOUCH = 0xD0,
+   CLAP_MIDIINFO_STATUS_PITCHBEND = 0xE0,
+
+   //CLAP note expressions.
+   //A host may map MIDI messages to note expressions, so it needs to know which note expressions are supported.
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_VOLUME = 0x100,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_PAN = 0x101,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_TUNING = 0x102,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_VIBRATO = 0x103,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_EXPRESSION = 0x104,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_BRIGHTNESS = 0x105,
+   CLAP_MIDIINFO_STATUS_NOTE_EXPRESSION_PRESSURE = 0x106

```

Comment:
wouldn't this expression of which NE you support be better in a different extension? We've replicated the same thing into the VST wrapper so seeing it here makes me think a 'note expression capabilities' separate extension may be more applicable?

Reply:
Tagging @defiantnerd on this comment too.

Reply:
Would be fine with me.